spec/spec.md

```diff
@@ -0,0 +1,2604 @@
+    (see section [8](#flows) for an explanation of flow).
+
+4.  Either the flow of the right-hand side expression is source or
+    duplex, or the right-hand side expression has a passive type.
+
+Connect statements from a narrower ground type component to a wider
+ground type component will have its value automatically sign-extended or
+zero-extended to the larger bit width. The behaviour of connect
+statements between two circuit components with aggregate types is
+defined by the connection algorithm in section
+[5.1.1](#connection_algorithm){reference-type="ref"
+reference="connection_algorithm"}.
+
+### The Connection Algorithm {#connection_algorithm}
+
+Connect statements between ground types cannot be expanded further.
+
+Connect statements between two vector typed components recursively
+connects each sub-element in the right-hand side expression to the
+corresponding sub-element in the left-hand side expression.
+
+Connect statements between two bundle typed components connects the i'th
+field of the right-hand side expression and the i'th field of the
+left-hand side expression. If the i'th field is not flipped, then the
+right-hand side field is connected to the left-hand side field.
+Conversely, if the i'th field is flipped, then the left-hand side field
+is connected to the right-hand side field.
+
+## Partial Connects {#partial_connects}
+
+Like the connect statement, the partial connect statement is also used
+to specify a physically wired connection between two circuit components.
+However, it enforces fewer restrictions on the types and widths of the
+circuit components it connects.
+
+In order for a partial connect to be legal the following conditions must
+hold:
+
+1.  The types of the left-hand and right-hand side expressions must be
+    weakly equivalent (see section
+    [4.6](#weak_type_equivalence){reference-type="ref"
+    reference="weak_type_equivalence"} for details).
+
+2.  The flow of the left-hand side expression must be sink or duplex
+    (see section [\[flow\]](#flow){reference-type="ref"
+    reference="flow"} for an explanation of flow).
+
+3.  Either the flow of the right-hand side expression is source or
+    duplex, or the right-hand side expression has a passive type.
+
+Partial connect statements from a narrower ground type component to a
+wider ground type component will have its value automatically
+sign-extended to the larger bit width. Partial connect statements from a
+wider ground type component to a narrower ground type component will
+have its value automatically truncated to fit the smaller bit width.
+
+Intuitively, bundle fields with matching names will be connected
+appropriately, while bundle fields not present in both types will be
+ignored. Similarly, vectors with mismatched lengths will be connected up
+to the shorter length, and the remaining sub-elements are ignored. The
+full algorithm is detailed in section
+[5.2.1](#partial_connection_algorithm){reference-type="ref"
```

Comment:
weird links.

Reply:
This is good to go now. The links will now show up and use the original link color (blue).

LaTeX spec:

<img width="829" alt="Screen Shot 2022-02-16 at 01 22 24" src="https://user-images.githubusercontent.com/1018530/154208027-fb44b86d-71b6-43e5-98a2-689b24947cc2.png">

Markdown spec:

<img width="831" alt="Screen Shot 2022-02-16 at 01 22 50" src="https://user-images.githubusercontent.com/1018530/154208072-c59ebf22-a067-47e1-b555-ce6180876333.png">